x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx

```diff
@@ -0,0 +1,159 @@
+
+    setEnginesData(params, callbacks);
+  }, [enginesPage]);
+
+  useEffect(() => {
+    if (hasPlatinumLicense(license)) {
+      const params = { type: 'meta', pageIndex: metaEnginesPage };
+      const callbacks = { setResults: setMetaEngines, setResultsTotal: setMetaEnginesTotal };
+
+      setEnginesData(params, callbacks);
+    }
+  }, [license, metaEnginesPage]);
+
+  if (hasErrorConnecting) return <ErrorState />;
+  if (isLoading) return <LoadingState />;
+  if (!engines.length) return <EmptyState />;
+
+  return (
+    <EuiPage restrictWidth className="engineOverview">
+      <SetBreadcrumbs isRoot />
+      <SendTelemetry action="viewed" metric="engines_overview" />
+
+      <EuiPageBody>
+        <EngineOverviewHeader />
+
+        <EuiPageContent panelPaddingSize="s" className="engineOverview__body">
+          <EuiPageContentHeader>
+            <EuiTitle size="s">
+              <h2>
+                <img src={EnginesIcon} alt="" className="engineIcon" />
+                <FormattedMessage
+                  id="xpack.enterpriseSearch.appSearch.enginesOverview.engines"
+                  defaultMessage="Engines"
+                />
+              </h2>
+            </EuiTitle>
+          </EuiPageContentHeader>
+          <EuiPageContentBody data-test-subj="appSearchEngines">
+            <EngineTable
+              data={engines}
+              pagination={{
+                totalEngines: enginesTotal,
+                pageIndex: enginesPage - 1,
+                onPaginate: setEnginesPage,
+              }}
+            />
+          </EuiPageContentBody>
+
+          {metaEngines.length > 0 && (
+            <>
+              <EuiSpacer size="xl" />
+              <EuiPageContentHeader>
+                <EuiTitle size="s">
+                  <h2>
+                    <img src={MetaEnginesIcon} alt="" className="engineIcon" />
+                    <FormattedMessage
+                      id="xpack.enterpriseSearch.appSearch.enginesOverview.metaEngines"
+                      defaultMessage="Meta Engines"
+                    />
+                  </h2>
+                </EuiTitle>
+              </EuiPageContentHeader>
+              <EuiPageContentBody data-test-subj="appSearchMetaEngines">
+                <EngineTable
+                  data={metaEngines}
+                  pagination={{
+                    totalEngines: metaEnginesTotal,
+                    pageIndex: metaEnginesPage - 1,
+                    onPaginate: setMetaEnginesPage,
+                  }}
+                />
+              </EuiPageContentBody>
+            </>
+          )}
+        </EuiPageContent>
+      </EuiPageBody>
+    </EuiPage>
+  );
+};
+
+/**
+ * Type definitions
+ */
+
+interface IGetEnginesParams {
+  type: string;
+  pageIndex: number;
+}
+interface ISetEnginesCallbacks {
+  setResults: React.Dispatch<React.SetStateAction<never[]>>;
+  setResultsTotal: React.Dispatch<React.SetStateAction<number>>;
+}
```

Comment:
Minor nit, everywhere else interfaces are defined above the components and the comment is redundant and could be removed.

Reply:
I can't super remember but I think I originally moved them down there because they were super long (this was before a refactor that cleaned them up), and I didn't want to scroll past it 😂 will definitely move them back up! 👍